packages/devtools_app/lib/src/app.dart

```diff
@@ -475,6 +494,25 @@ class DevToolsAppState extends State<DevToolsApp> with AutoDisposeMixin {
       screens.removeWhere((s) => s is! ExtensionScreen);
     }
   }
+
+  String? _screenUnavailableReason(String screenId) {
+    final screenMetaData = ScreenMetaData.lookup(screenId);
+    String? disabledReason;
+    if (screenMetaData != null) {
+      final screen =
+          _originalScreens.firstWhere((s) => s.screenId == screenMetaData.id);
+      final reason = shouldShowScreen(screen).disabledReason;
+      if (reason == ScreenDisabledReason.requiresDartLibrary) {
+        // Special case for screens that require a library since the message
+        // needs to be generated dynamically.
+        disabledReason = 'The ${screen.title} screen requires library '
+            '${screen.requiresLibrary}, but the library was not detected.';
+      } else if (reason?.message != null) {
+        disabledReason = 'The ${screen.title} screen ${reason!.message!}';
+      }
```

Comment:
Not important, but FWIW using `case String` is a nice way to avoid the `!`s here:

```dart
if (reason?.message case String message) {
  disabledReason = 'The ${screen.title} screen $message';
}
```

Reply:
Nice! will add to a follow up PR.

Reply:
Oh and you can avoid writing the type (`String`) again like this:

```dart
if (reason?.message case final message?) {
```